src/validation_workflow/validation.py

```diff
@@ -38,6 +41,14 @@ def copy_artifact(self, filepath: str, tempdir_path: str) -> bool:
         else:
             raise Exception("Provided path for local artifacts does not exist")
 
+    def check_for_security_plugin(self, work_dir: str, distribution: str) -> bool:
+        list_cmd = "dir" if distribution == "zip" else "ls"
+        (_, plugins_list, _) = execute(f"{list_cmd} plugins", work_dir, True, False)
+        return "opensearch-security" in plugins_list
+
+    def get_version(self, project: str) -> str:
+        return re.search(r'(\d+\.\d+\.\d+)', os.path.basename(project)).group(1)
```

Comment:
What's this for?

Reply:
Hey @peterzhuamazon. This will cover the case when only url's are provided as input, in all other cases version is provide as an argument. It will fetch the version from the url which will used in get_password(version) method.

Reply:
Is Regex here better than just do a split and check the fields?

Reply:
After discussion, @Divyaasm confirms we can keep this implementation for now.